process_report/process_report.py

```diff
@@ -232,16 +234,32 @@ def main():
         nonbillable_projects=projects,
     )
 
+    ### Remove nonbillables
```

Comment:
Ideally, we want all Processor steps to happen before the Invoices steps. Is this not possible here?

Reply:
With the way I'm writing the `RemoveNonbillables` Processor, the nonbillable projects are completely removed from the dataframe, so it would not be possible to perform this processing step before creating the Nonbillable-Invoice. A similar logic applies to `ValidateBillablePIsProcessor`.

I could have modified `RemoveNonbillables` so that instead of removing the nonbillable projects outright, it would add a column to "mark" which projects were non-billable. I just figured my current approach would result in the simplest diff.

Do you think I should still amend the PR so that all Processor steps happen before Invoice steps?

Reply:
@QuanMPhm I can see the advantage of having an `Is Billable?` (Y/N) column. Give it a shot and see how easily you can implement it. My concern is that a lot of the processing would then require checking for `Is Billable? = Y` before applying it, such as credits, etc.

I'll think about this a tiny bit more in the meanwhile. If you don't hear from me within Thursday ping me on Slack. Sometimes I lose track of these things.